File: app/src/main/java/com/sakuna63/tumbin/application/di/component/ActivityComponent.java
package com.sakuna63.tumbin.application.di.component;

import com.sakuna63.tumbin.application.activity.PostActivity;
import com.sakuna63.tumbin.application.di.module.ActivityModule;
import com.sakuna63.tumbin.application.di.scope.ActivityScope;
import com.sakuna63.tumbin.data.api.TumblrService;

import dagger.Component;
import io.realm.RealmConfiguration;

@ActivityScope
@Component(dependencies = ApplicationComponent.class, modules = ActivityModule.class)
public interface ActivityComponent {
    void inject(PostActivity activity);

    RealmConfiguration realmConfiguration();

    TumblrService tumblrService();
}

File: app/src/main/java/com/sakuna63/tumbin/application/di/component/ActivityComponent.kt
package com.sakuna63.tumbin.application.di.component

import com.sakuna63.tumbin.application.activity.DashboardPostActivity
import com.sakuna63.tumbin.application.di.module.ActivityModule
import com.sakuna63.tumbin.application.di.module.FragmentModule
import com.sakuna63.tumbin.application.di.scope.ActivityScope
import dagger.Subcomponent
import io.realm.RealmConfiguration

@ActivityScope
@Subcomponent(modules = arrayOf(ActivityModule::class))
interface ActivityComponent {
    fun plus(module: FragmentModule): FragmentComponent

    fun inject(activity: DashboardPostActivity)

    fun realmConfiguration(): RealmConfiguration
}
